staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -74,30 +72,39 @@ type Patcher struct {
 	OpenapiSchema openapi.Resources
 }
 
-func newPatcher(o *ApplyOptions, info *resource.Info) (*Patcher, error) {
+func newPatcher(o *ApplyOptions, info *resource.Info, helper *resource.Helper) (*Patcher, error) {
 	var openapiSchema openapi.Resources
 	if o.OpenAPIPatch {
 		openapiSchema = o.OpenAPISchema
 	}
 
 	return &Patcher{
 		Mapping:       info.Mapping,
-		Helper:        resource.NewHelper(info.Client, info.Mapping),
-		DynamicClient: o.DynamicClient,
+		Helper:        helper,
 		Overwrite:     o.Overwrite,
 		BackOff:       clockwork.NewRealClock(),
 		Force:         o.DeleteOptions.ForceDeletion,
 		Cascade:       o.DeleteOptions.Cascade,
 		Timeout:       o.DeleteOptions.Timeout,
 		GracePeriod:   o.DeleteOptions.GracePeriod,
-		ServerDryRun:  o.DryRunStrategy == cmdutil.DryRunServer,
 		OpenapiSchema: openapiSchema,
 		Retries:       maxPatchRetry,
 	}, nil
 }
 
 func (p *Patcher) delete(namespace, name string) error {
-	return runDelete(namespace, name, p.Mapping, p.DynamicClient, p.Cascade, p.GracePeriod, p.ServerDryRun)
+	options := metav1.DeleteOptions{}
```

Comment:
I'm not sure we should do this. The exact same code is run by the prune, so we've duplicated the code (which is why there is a common `runDelete()` function. Let me know what you think.

Reply:
I've added a method in prune to convert some delete options to `metav1.DeleteOptions`, while still removing the dynamic client from the patcher, which is the most important part. Could you take a look?